.github/workflows/snapshot-ci.yml

```diff
@@ -0,0 +1,267 @@
+        python:
+          - { name: cp38, version: '3.8' }
+          - { name: cp39, version: '3.9' }
+          - { name: cp310, version: '3.10' }
+          - { name: cp311, version: '3.11' }
+          - { name: cp312, version: '3.12' }
+
+      fail-fast: false
+
+    steps:
+
+      #SETUP JDK
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin'
+          java-version: '17'
+
+      #SETUP PYTHON
+      - name: Set up Python ${{ matrix.python.version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python.version }}
+      - name: Install Python dependencies
+        run: python -m pip install --upgrade pip
+
+      #SETUP GRAALVM
+      - name: Setup GraalVM
+        uses: graalvm/setup-graalvm@v1.1.5
+        with:
+          java-version: '17'
+          distribution: 'graalvm'
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+
+      #DEFINE SCRIPTS PATH
+      - name: Set up script path
+        shell: bash
+        run: |
+          SCRIPTS_PATH="${GITHUB_WORKSPACE}/scripts/.github/workflows/scripts"
+          if [[ "${{ matrix.config.name }}" == "windows" ]]; then
+            SCRIPTS_PATH=$(echo "$SCRIPTS_PATH" | sed 's/\\/\//g')
+          fi
+          echo "SCRIPTS_PATH=$SCRIPTS_PATH" >> $GITHUB_ENV  
+
+      #BUILD CORE
+      - name: Checkout core-sources
+        uses: actions/checkout@v4
+        with:
+          repository: powsybl/powsybl-core
+          ref: main
+          path: powsybl-core
+      - name: Build CORE
+        run: mvn -batch-mode --no-transfer-progress clean install -DskipTests
+        working-directory: ./powsybl-core
+        shell: bash
+      - name: get CORE_VERSION
+        run: echo "CORE_VERSION=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)" >> $GITHUB_ENV
+        working-directory: ./powsybl-core
+        shell: bash
+
+      #CHECKOUT SCRIPT
+      - name: checkout script
+        uses: actions/checkout@v4
+        with:
+          sparse-checkout: |
+            .github
+          sparse-checkout-cone-mode: false
+          path: scripts
+
+      #BUILD LOADFLOW
+      - name: Check LOADFLOW SNAPSHOT branch
+        run : ${{ env.SCRIPTS_PATH }}/check_snapshot_branch.sh "https://github.com/powsybl/powsybl-open-loadflow.git" ${{ env.CORE_VERSION }}
+        shell: bash
+      - name: Checkout loadflow-sources
+        uses: actions/checkout@v4
+        with:
+          repository: powsybl/powsybl-open-loadflow
+          ref: ${{ env.SNAPSHOT_BRANCH }}
+          path: powsybl-open-loadflow
+      - name: Change core version
+        run: mvn versions:set-property -Dproperty=powsybl-core.version -DnewVersion=${{ env.CORE_VERSION}} -DgenerateBackupPoms=false
+        shell: bash
+        working-directory: ./powsybl-open-loadflow
+      - name: Build LOADFLOW
+        run: mvn -batch-mode --no-transfer-progress clean install -DskipTests
```

Comment:
General comment for all repos: maybe we should keep the unit tests run to avoid building a wheel based in code that do not pass tests ?